pymc4/model/base.py

```diff
@@ -68,14 +68,41 @@ def get_mode(state, rv, *args, **kwargs):
             returns = self.session.run(list(values_collector.result.values()))
         return dict(zip(values_collector.result.keys(), returns))
 
-    def target_log_prob_fn(self, *args, **kwargs):
+    def log_prob_fn(self, x,  *args, **kwargs):
         logp = 0
         for i in self.unobserved.keys():
-            print(kwargs.get(i))
             logp += self.unobserved[i].rv.distribution.log_prob(value=kwargs.get(i))
 
         return logp
 
+    def target_log_prob_fn(self, *args, **kwargs):
+        """
+        Unnormalized target density as a function of unobserved states.
+        """
+
+        def log_joint_fn(*args, **kwargs):
+            states = dict(zip(self.unobserved.keys(), args))
+            states.update(self.observed)
+            log_probs = []
```

Comment:
https://github.com/pymc-devs/pymc4/blob/functional/pymc4/util/interceptors.py#L110
```python
collect_log_prob = CollectLogProb(states)
with ed.interception(collect_log_prob):
    self._f(self._cfg)
return collect_log_prob.result
```

Reply:
changing it, was facing problems with states before. Now working.

Reply:
@ferrine `interceptors.CollectLogProb` only works with model like
``` python
@model.define
def process(cfg=None):
    mu = ed.Normal(0., 1., name="mu")
    obs = ed.Normal(0., 1., name="obs")
    return obs
```
and not model like
``` python
@model.define
def process(cfg=None):
    mu = ed.Normal(0., 1., name="mu")
    obs = ed.Normal(mu, 1., name="obs")
    return obs
```

Reply:
hmm, what's happening?